temporalio/converter.py

```diff
@@ -677,6 +679,21 @@ def encode_search_attribute_values(
     return default().payload_converter.to_payloads([safe_vals])[0]
 
 
+def get_iso_datetime_parser() -> Callable[[str], datetime]:
+    """Isolates system version check and returns relevant datetime passer
+
+    Returns:
+        A callable to parse date strings into datetimes.
+    """
+    if sys.version_info >= (3, 11):
+        return datetime.fromisoformat  # noqa
+    else:
+        # Isolate import for py > 3.11, as dependency only installed for < 3.11
+        from dateutil import parser
+
+        return parser.isoparse
```

Comment:
Can we put:

```python
if sys.version_info < (3, 11):
  import dateutil.parser
```

Near the top of the file to keep imports together up there and then down here just:

```suggestion
    if sys.version_info < (3, 11):
        return dateutil.parser.isoparse
    return datetime.fromisoformat
```